modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultConstantFoldingOptimizationPhase.java

```diff
@@ -621,22 +621,19 @@ public void visitStringConcatenation(StringConcatenationNode irStringConcatenati
             int j = i;
             irRightNode.visit(this, (e) -> irStringConcatenationNode.getArgumentNodes().set(j + 1, e));
 
-            if (irLeftNode instanceof ConstantNode && irRightNode instanceof ConstantNode) {
-                ConstantNode irConstantNode = (ConstantNode) irLeftNode;
+            if (irLeftNode instanceof ConstantNode irConstantNode && irRightNode instanceof ConstantNode) {
                 irConstantNode.attachDecoration(
                     new IRDConstant(
                         "" + irConstantNode.getDecorationValue(IRDConstant.class) + irRightNode.getDecorationValue(IRDConstant.class)
                     )
                 );
                 irConstantNode.attachDecoration(new IRDExpressionType(String.class));
                 irStringConcatenationNode.getArgumentNodes().remove(i + 1);
-            } else if (irLeftNode instanceof NullNode && irRightNode instanceof ConstantNode) {
-                ConstantNode irConstantNode = (ConstantNode) irRightNode;
+            } else if (irLeftNode instanceof NullNode && irRightNode instanceof ConstantNode irConstantNode) {
```

Comment:
Same as above, seems simpler to just use `irRightNode` in the block? Also easier to read.